extensions/flyway/deployment/src/test/java/io/quarkus/flyway/test/FlywayExtensionConfigMultiDataSourcesTest.java

```diff
@@ -0,0 +1,75 @@
+package io.quarkus.flyway.test;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.flywaydb.core.Flyway;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.flyway.FlywayDataSource;
+import io.quarkus.test.QuarkusUnitTest;
+
+/**
+ * Test a full configuration with default and two named datasources plus their flyway settings.
+ */
+public class FlywayExtensionConfigMultiDataSourcesTest {
+
+    @Inject
+    FlywayExtensionConfigFixture fixture;
+
+    // Quarkus built objects
+    @Inject
+    Flyway flyway;
+
+    @Inject
+    @FlywayDataSource("users")
```

Comment:
You said people should use `@Named`. Is it tested somewhere?

Reply:
Only my opinion. And you're right. I only did this in "FlywayExtensionConfigMultiDataSourcesTest". I will let this test do all injection variations and use `@Named` for all other tests.

Reply:
I added two static factory methods and refactored https://github.com/quarkusio/quarkus/pull/3521#discussion_r352529516